compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -17,7 +18,9 @@ use rustc_session::lint;
 use rustc_session::parse::feature_err;
 use rustc_span::symbol::Ident;
 use rustc_span::{sym, Span};
+use rustc_target::abi::VariantIdx;
 use rustc_target::spec::{abi, SanitizerSet};
+use rustc_type_ir::inherent::*;
```

Comment:
wait omfg please don't import this -- this is only meant to be used in the trait solver

Reply:
Any trait you're using from `rustc_type_ir::inherent::*` should have an equivalent method *not* in that struct.

Reply:
Sorry -- I'll also make sure we implement a lint to disallow using these outside of the new solver and type IR crates.